Agent crashes on shutdown when no Loki config is present

To reproduce, use either of the example configs, and then shut down the Agent:

```
level=error ts=2020-08-12T17:32:10.600369Z caller=main.go:46 msg="failed to create loki log collection instance" err="at least one client config should be provided"
[...]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x21b8f64]

goroutine 1 [running]:
github.com/grafana/agent/pkg/loki.(*Loki).Stop(...)
        /home/robert/dev/grafana/agent/pkg/loki/loki.go:53
main.main()
        /home/robert/dev/grafana/agent/cmd/agent/main.go:85 +0x514
exit status 2
```

We probably shouldn't be starting Promtail if it's not configured at all.

--- a/cmd/agent/main.go
+++ b/cmd/agent/main.go
@@ -44,6 +44,7 @@ func main() {
 	lokiLogs, err := loki.New(cfg.Loki, util.Logger)
 	if err != nil {
 		level.Error(util.Logger).Log("msg", "failed to create loki log collection instance", "err", err)
+		os.Exit(1)
 	}
 
 	srv, err := server.New(cfg.Server)

--- a/pkg/loki/loki.go
+++ b/pkg/loki/loki.go
@@ -5,6 +5,7 @@ import (
 	"flag"
 
 	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
 	"github.com/grafana/loki/pkg/promtail"
 	"github.com/grafana/loki/pkg/promtail/client"
 	"github.com/grafana/loki/pkg/promtail/config"
@@ -35,6 +36,11 @@ type Loki struct {
 func New(c Config, l log.Logger) (*Loki, error) {
 	l = log.With(l, "component", "loki")
 
+	if len(c.ClientConfigs) == 0 {
+		level.Info(l).Log("msg", "skipping creation of a promtail because no client_configs are present")
+		return &Loki{}, nil
+	}
+
 	p, err := promtail.New(config.Config{
 		ServerConfig:    server.Config{Disable: true},
 		ClientConfigs:   c.ClientConfigs,
@@ -50,5 +56,8 @@ func New(c Config, l log.Logger) (*Loki, error) {
 }
 
 func (l *Loki) Stop() {
-	l.p.Shutdown()
+	// l.p will be nil when there weren't any client_configs set.
+	if l.p != nil {
+		l.p.Shutdown()
+	}
 }

